src/foundry/foundry.js/roll.d.ts

```diff
@@ -1,501 +1,505 @@
-   *                      (default: `{}`)
-   * @param rollMode    - The template roll mode to use for the message from CONFIG.Dice.rollModes
-   * @param create      - Whether to automatically create the chat message, or only return the
-   *                      prepared chatData object.
-   *                      (default: `true`)
-   * @returns A promise which resolves to the created ChatMessage entity, if create is true
-   *          or the Object of prepared chatData otherwise.
-   */
-  toMessage<T extends DeepPartial<ChatMessage.CreateData> = {}>(
-    messageData?: T,
-    { rollMode, create }?: { rollMode?: foundry.CONST.DiceRollMode; create?: true }
-  ): Promise<ChatMessage>;
-  toMessage<T extends DeepPartial<ChatMessage.CreateData> = {}>(
-    messageData: T,
-    { rollMode, create }: { rollMode?: foundry.CONST.DiceRollMode; create: false }
-  ): Roll.MessageData<T>;
-  toMessage<T extends DeepPartial<ChatMessage.CreateData> = {}>(
-    messageData: T,
-    { rollMode, create }: { rollMode?: foundry.CONST.DiceRollMode; create: boolean }
-  ): Promise<ChatMessage> | Roll.MessageData<T>;
+    /**
+     * Recreate a Roll instance using a provided data object
+     * @param data - Unpacked data representing the Roll
+     * @returns A reconstructed Roll instance
+     */
+    static fromData<T extends Roll>(this: ConstructorOf<T>, data: Data): T;
 
-  /**
-   * Represent the data of the Roll as an object suitable for JSON serialization.
-   * @returns Structured data which can be serialized into JSON
-   */
-  toJSON(): {
-    class: string;
-    options: Roll.Options;
-    dice: DiceTerm[];
-    formula: string;
-    terms: RollTerm[];
-    total: number | null;
-    evaluated: boolean;
-  };
+    /**
+     * Recreate a Roll instance using a provided JSON string
+     * @param json - Serialized JSON data representing the Roll
+     * @returns A reconstructed Roll instance
+     */
+    static fromJSON(json: string): Roll;
 
-  /**
-   * Recreate a Roll instance using a provided data object
-   * @param data - Unpacked data representing the Roll
-   * @returns A reconstructed Roll instance
-   */
-  static fromData<T extends Roll>(this: ConstructorOf<T>, data: Roll.Data): T;
+    /**
+     * Manually construct a Roll object by providing an explicit set of input terms
+     * @param terms -The array of terms to use as the basis for the Roll
+     * @param options - Additional options passed to the Roll constructor
+     * @returns The constructed Roll instance
+     *
+     * @example
+     * ```typescript
+     * const t1 = new Die({number: 4, faces: 8};
+     * const plus = new OperatorTerm({operator: "+"});
+     * const t2 = new NumericTerm({number: 8});
+     * const roll = Roll.fromTerms([t1, plus, t2]);
+     * roll.formula; // 4d8 + 8
+     * ```
+     */
+    static fromTerms<D extends Record<string, unknown>>(terms: RollTerm[], options?: Partial<Options>): Roll<D>;
```

Comment:
If I understand correctly, the `data` of the created roll will always be `{}`, so this method should not be generic but instead should return `Roll<{}>` (or even better, pull the used roll class from the `CONFIG` because that's what `create` does).